File: CS/NavigationDemo.Wpf/ViewModels/AutoTraderViewModel.cs
using System.Collections.Generic;
using System.Linq;
using DevExpress.DemoData.Models.Vehicles;
using DevExpress.Mvvm.POCO;
using DevExpress.Utils.Filtering;

namespace NavigationDemo {
    public class AutoTraderViewModel {
        public IEnumerable<Model> Vehicles { get; private set; }
        public IEnumerable<Trademark> Trademarks { get; private set; }
        public IEnumerable<TransmissionType> TransmissionTypes { get; private set; }
        public IEnumerable<BodyStyle> BodyStyles { get; private set; }
        public int[] DoorTypes { get; private set; }
        public decimal MinPrice { get; private set; }
        public decimal MaxPrice { get; private set; }
        public int MinMPGCity { get; private set; }
        public int MaxMPGCity { get; private set; }
        public int MinMPGHighway { get; private set; }
        public int MaxMPGHighway { get; private set; }
        public AutoTraderViewModel() {
            if(this.IsInDesignMode()) return;
            var context = new VehiclesContext();
            Vehicles = context.Models.ToList();
            Trademarks = context.Trademarks.ToList();
            TransmissionTypes = context.TransmissionTypes.ToList();
            BodyStyles = context.BodyStyles.ToList();
            DoorTypes = new int[] { 2, 3, 4 };
            CalculateProperties();
            context.Dispose();
        }
        void CalculateProperties() {
            MinPrice = Vehicles.Min(x => x.Price);
            MaxPrice = Vehicles.Max(x => x.Price);
            MinMPGCity = Vehicles.Min(x => x.MPGCity ?? int.MaxValue);
            MaxMPGCity = Vehicles.Max(x => x.MPGCity ?? int.MinValue);
            MinMPGHighway = Vehicles.Min(x => x.MPGHighway ?? int.MaxValue);
            MaxMPGHighway = Vehicles.Max(x => x.MPGHighway ?? int.MinValue);
        }
    }

    public class VehiclesFilteringViewModel {
        [FilterRange("MinPrice", "MaxPrice")]
        public decimal Price { get; set; }
        [FilterLookup("Trademarks", DisplayMember = "Name", ValueMember = "ID")]
        public long TrademarkID { get; set; }
        [FilterLookup("TransmissionTypes", DisplayMember = "Name", ValueMember = "ID")]
        public long TransmissionTypeID { get; set; }
        [FilterLookup("BodyStyles", DisplayMember = "Name", ValueMember = "ID")]
        public long BodyStyleID { get; set; }
        [FilterLookup("DoorTypes")]
        public int Doors { get; set; }
        [FilterRange("MinMPGCity", "MaxMPGCity")]
        public int? MPGCity { get; set; }
        [FilterRange("MinMPGHighway", "MaxMPGHighway")]
        public int? MPGHighway { get; set; }
    }
}

File: VB/NavigationDemo.Wpf/ViewModels/AutoTraderViewModel.vb
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.DemoData.Models.Vehicles
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Utils.Filtering

Namespace NavigationDemo
    Public Class AutoTraderViewModel
        Private privateVehicles As IEnumerable(Of Model)
        Public Property Vehicles() As IEnumerable(Of Model)
            Get
                Return privateVehicles
            End Get
            Private Set(ByVal value As IEnumerable(Of Model))
                privateVehicles = value
            End Set
        End Property
        Private privateTrademarks As IEnumerable(Of Trademark)
        Public Property Trademarks() As IEnumerable(Of Trademark)
            Get
                Return privateTrademarks
            End Get
            Private Set(ByVal value As IEnumerable(Of Trademark))
                privateTrademarks = value
            End Set
        End Property
        Private privateTransmissionTypes As IEnumerable(Of TransmissionType)
        Public Property TransmissionTypes() As IEnumerable(Of TransmissionType)
            Get
                Return privateTransmissionTypes
            End Get
            Private Set(ByVal value As IEnumerable(Of TransmissionType))
                privateTransmissionTypes = value
            End Set
        End Property
        Private privateBodyStyles As IEnumerable(Of BodyStyle)
        Public Property BodyStyles() As IEnumerable(Of BodyStyle)
            Get
                Return privateBodyStyles
            End Get
            Private Set(ByVal value As IEnumerable(Of BodyStyle))
                privateBodyStyles = value
            End Set
        End Property
        Private privateDoorTypes As Integer()
        Public Property DoorTypes() As Integer()
            Get
                Return privateDoorTypes
            End Get
            Private Set(ByVal value As Integer())
                privateDoorTypes = value
            End Set
        End Property
        Private privateMinPrice As Decimal
        Public Property MinPrice() As Decimal
            Get
                Return privateMinPrice
            End Get
            Private Set(ByVal value As Decimal)
                privateMinPrice = value
            End Set
        End Property
        Private privateMaxPrice As Decimal
        Public Property MaxPrice() As Decimal
            Get
                Return privateMaxPrice
            End Get
            Private Set(ByVal value As Decimal)
                privateMaxPrice = value
            End Set
        End Property
        Private privateMinMPGCity As Integer
        Public Property MinMPGCity() As Integer
            Get
                Return privateMinMPGCity
            End Get
            Private Set(ByVal value As Integer)
                privateMinMPGCity = value
            End Set
        End Property
        Private privateMaxMPGCity As Integer
        Public Property MaxMPGCity() As Integer
            Get
                Return privateMaxMPGCity
            End Get
            Private Set(ByVal value As Integer)
                privateMaxMPGCity = value
            End Set
        End Property
        Private privateMinMPGHighway As Integer
        Public Property MinMPGHighway() As Integer
            Get
                Return privateMinMPGHighway
            End Get
            Private Set(ByVal value As Integer)
                privateMinMPGHighway = value
            End Set
        End Property
        Private privateMaxMPGHighway As Integer
        Public Property MaxMPGHighway() As Integer
            Get
                Return privateMaxMPGHighway
            End Get
            Private Set(ByVal value As Integer)
                privateMaxMPGHighway = value
            End Set
        End Property
        Public Sub New()
            If Me.IsInDesignMode() Then
                Return
            End If
            Dim context = New VehiclesContext()
            Vehicles = context.Models.ToList()
            Trademarks = context.Trademarks.ToList()
            TransmissionTypes = context.TransmissionTypes.ToList()
            BodyStyles = context.BodyStyles.ToList()
            DoorTypes = New Integer() { 2, 3, 4 }
            CalculateProperties()
            context.Dispose()
        End Sub
        Private Sub CalculateProperties()
            MinPrice = Vehicles.Min(Function(x) x.Price)
            MaxPrice = Vehicles.Max(Function(x) x.Price)
            MinMPGCity = Vehicles.Min(Function(x) If(x.MPGCity, Integer.MaxValue))
            MaxMPGCity = Vehicles.Max(Function(x) If(x.MPGCity, Integer.MinValue))
            MinMPGHighway = Vehicles.Min(Function(x) If(x.MPGHighway, Integer.MaxValue))
            MaxMPGHighway = Vehicles.Max(Function(x) If(x.MPGHighway, Integer.MinValue))
        End Sub
    End Class

    Public Class VehiclesFilteringViewModel
        <FilterRange("MinPrice", "MaxPrice")>
        Public Property Price() As Decimal
        <FilterLookup("Trademarks", DisplayMember := "Name", ValueMember := "ID")>
        Public Property TrademarkID() As Long
        <FilterLookup("TransmissionTypes", DisplayMember := "Name", ValueMember := "ID")>
        Public Property TransmissionTypeID() As Long
        <FilterLookup("BodyStyles", DisplayMember := "Name", ValueMember := "ID")>
        Public Property BodyStyleID() As Long
        <FilterLookup("DoorTypes")>
        Public Property Doors() As Integer
        <FilterRange("MinMPGCity", "MaxMPGCity")>
        Public Property MPGCity() As Integer?
        <FilterRange("MinMPGHighway", "MaxMPGHighway")>
        Public Property MPGHighway() As Integer?
    End Class
End Namespace
